includes/class-search.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * ActivityPub Search Enhancement Class
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub;
+
+use Activitypub\Collection\Interactions;
+
+/**
+ * ActivityPub Search Enhancement Class
+ *
+ * This class enhances WordPress search functionality to detect URLs in search queries
+ * and attempt to import ActivityPub objects if found, otherwise falls back to classic search.
+ */
+class Search {
+
+	/**
+	 * Initialize the search enhancement.
+	 */
+	public static function init() {
+		\add_filter( 'pre_get_posts', array( self::class, 'enhance_public_search' ) );
+		\add_action( 'admin_init', array( self::class, 'enhance_admin_comment_search' ) );
+	}
+
+	/**
+	 * Enhance public search functionality to check for URLs and ActivityPub objects.
+	 *
+	 * @param \WP_Query $query The WP_Query instance.
+	 * @return \WP_Query The modified query.
+	 */
+	public static function enhance_public_search( $query ) {
+		// Check for a valid user session.
+		if ( ! is_user_logged_in() ) {
```

Comment:
This should probably be a capability check, so not any logged-in user can do that. `moderate_comments` or something? Same for the other callback.

Reply:
I'd err on the side of being too strict than not enough.

Reply:
done!